### PR TITLE
Represent UUIDs as String

### PR DESCRIPTION
**Pull request by shawnhathaway**
**What changed?**
Change UUIDs represented as `[]byte` to `string` in:

- Go interfaces
- PersistenceBlob protos
- HistoryBranchToken proto

**Why?**
Not enough time to create a low-risk abstraction layer between user consumption of UUIDs and our machine consumption. Binary data is primarily printed by base64 and we have no control in some areas like json. Keeping this as string for now for stability at expense of keeping all IDs UTF-8 encoded in memory and disk. 

**How did you test it?**
Tests passing locally + Buidkite

**Potential risks**
Large type conversion that is server wide, possible logging regressions or parsing errors from CLI.

### CODE REVIEW

File: common/namespace/handler_GlobalNamespaceEnabled_NotMasterCluster_test.go [hunk 1]
```diff
@@ -575,7 +575,7 @@ func (s *namespaceHandlerGlobalNamespaceEnabledNotMasterClusterSuite) TestUpdate
 	_, err := s.MetadataManager.CreateNamespace(&persistence.CreateNamespaceRequest{
 		Namespace: &persistenceblobs.NamespaceDetail{
 			Info: &persistenceblobs.NamespaceInfo{
-				Id:          uuid.NewRandom(),
+		Id:          uuid.New(),
```
Comment (shawnhathaway) [inline common/namespace/handler_GlobalNamespaceEnabled_NotMasterCluster_test.go:578]: Will go fmt.

File: common/primitives/uuid.go [hunk 2]
```diff
@@ -49,6 +49,46 @@ func MustParseUUID(s string) UUID {
 	return u[:]
 }
 
+// ParseUUID returns a UUID parsed from the given string representation
```
Comment (shawnhathaway) [inline common/primitives/uuid.go:52]: Calling out new code as this is easy to miss in all the renames

File: common/namespace/transmissionTaskHandler_test.go [hunk 3]
```diff
@@ -73,7 +73,7 @@ func (s *transmissionTaskSuite) TearDownTest() {
 
 func (s *transmissionTaskSuite) TestHandleTransmissionTask_RegisterNamespaceTask_IsGlobalNamespace() {
 	taskType := replicationgenpb.ReplicationTaskType_NamespaceTask
-	id := primitives.NewUUID()
+	id := primitives.NewUUID().String()
```
Comment (shawnhathaway) [inline common/namespace/transmissionTaskHandler_test.go:76]: Used regexes to do most of this, hence different methods of string creation of UUIDs. 

I plan to follow up with an additional change to remove direct references to google/pborman UUID so `uuid.New()` and `uuid.NewRandom()` will instead use our `primitives.UUID`.

File: common/primitives/uuid.go [hunk 4]
```diff
@@ -49,6 +49,46 @@ func MustParseUUID(s string) UUID {
...
+// returns nil if the input is empty string
+// panics if the given input is malformed
```
Comment (markmark206) [inline common/primitives/uuid.go:54]: is it true? (nil if empty string and panic if malformed).

File: service/history/transferQueueActiveTaskExecutor.go [hunk 5]
```diff
@@ -358,16 +357,16 @@ func (t *transferQueueActiveTaskExecutor) processCancelExecution(
...
-	targetNamespaceEntry, err := t.shard.GetNamespaceCache().GetNamespaceByID(primitives.UUIDString(task.GetTargetNamespaceId()))
+	targetNamespaceEntry, err := t.shard.GetNamespaceCache().GetNamespaceByID(task.GetTargetNamespaceId())
 	if err != nil {
 		return err
 	}
 	targetNamespace := targetNamespaceEntry.GetInfo().Name
 
 	// handle workflow cancel itself
-	if bytes.Compare(task.GetNamespaceId(), task.GetTargetNamespaceId()) == 0 && task.GetWorkflowId() == task.GetTargetWorkflowId() {
+	if strings.Compare(task.GetNamespaceId(), task.GetTargetNamespaceId()) == 0 && task.GetWorkflowId() == task.GetTargetWorkflowId() {
```
Comment (markmark206) [inline service/history/transferQueueActiveTaskExecutor.go:367]: here and in other similar changes: a basic `task.GetNamespaceId() == task.GetTargetNamespaceId()` might be more readable?

**Comment by shawnhathaway** [comment]
> man, this is a monumental change. ; )

Regexes save me here :) 

> also, is there a reason we don't want to also change the types of ids in db schemas to string?

Yes, string UUID is 36 bytes vs 16 bytes in binary. Binary is highly preferable here and has no overhead as we need to parse the UUID at some point in persistence anyway.

File: common/namespace/handler_GlobalNamespaceEnabled_NotMasterCluster_test.go (hunk 1 above)
Comment (alexshtin) [reply common/namespace/handler_GlobalNamespaceEnabled_NotMasterCluster_test.go]: `make goimports`

**Pull request merged**